ld -shared  synaptics.o ps2comm.o eventcomm.o psmcomm.o alpscomm.o -o synaptics_drv.so
ld: synaptics.o: relocation R_X86_64_32S against `a local symbol' can not be used when making a shared object; recompile with -fPIC
synaptics.o: could not read symbols: Bad value
make: *** [synaptics_drv.so] Error 1
diff -Naur synaptics-0.14.5.orig/Makefile synaptics-0.14.5/Makefile
--- synaptics-0.14.5.orig/Makefile	2006-06-04 01:56:58.000000000 +0200
+++ synaptics-0.14.5/Makefile	2006-06-11 13:23:16.000000000 +0200
@@ -81,6 +81,9 @@
 CCOPTIONS := -pedantic -Wall -Wpointer-arith
 CCOPTIONS += $(call check_gcc,-fno-merge-constants,)
 CDEBUGFLAGS = -O2
+ifeq ($(ARCH),x86_64)
+	CDEBUGFLAGS += -fPIC
+endif
 CFLAGS = $(CDEBUGFLAGS) $(CCOPTIONS) $(ALLDEFINES) -DVERSION="\"$(VERSION)\"" -DVERSION_ID="$(VERSION_ID)"
 CFLAGSCLIENT = $(CDEBUGFLAGS) $(CCOPTIONS) -DVERSION="\"$(VERSION)\""  -DVERSION_ID="$(VERSION_ID)" -I$(X_INCLUDES_ROOT)/include
 
